### ISSUE TITLE
http headers get overridden on every request; unnecessary.

### ISSUE THREAD

**Issue by courtenay** [grantr/rubberband#45]
In http.rb we have this code;

```
  def request(method, operation, params={}, body=nil, headers={})
    begin
      response = @session.send(method, generate_uri(operation)) do |req|
        req.headers = headers
        req.params = params
        req.body = body
      end
```

Note that this forcibly overrides the headers of every request. However, the connect! method a few lines up does this:

```
  def connect!
    if @connect_block
      @session = Faraday.new :url => @server, :headers => {'User-Agent' => 'Rubberband'}, &@connect_block
```

That user-agent string will not get used, because headers are removed.

I'm actually sending http basic auth in the url (e.g. https://user:pass@1.2.3.4) which sets the correct headers (in the connect phase) but which get removed by the http request method. (my ES is set up behind a public-facing, http-basic-auth, ssl, haproxy).

So I have two solutions; one, remove the req.headers and the method param completely (they are never used!); or, req.headers = @session.headers.merge(headers) which would allow overriding of specific headers but wouldn't let you just clear them.

I'm happy to submit a patch, just wondering if there's a reason for the headers in there.

**Comment by grantr** [comment]
Christmas Eve? Really? ;)

Remind me to take a look at this on Wednesday!

**Comment by grantr** [comment]
Ok finally back. This might have broken when patron was replaced by faraday.

I don't think there's any good reason for the User-Agent header to be there. It can be removed. And I don't think elasticseach uses headers for its api, so there's little reason to accept headers in the request method.

Can you submit a patch with the way you'd like it to work?

**Comment by courtenay** [comment]
Pull Req #47 attached!

**Comment by grantr** [comment]
Released in 0.9.7. Thanks!
